neuro_flow/context_v2.py

```diff
@@ -0,0 +1,484 @@
+        if isinstance(job, ast.JobActionCall):
+            action_ast = await self._get_action_ast(job)
+            ctx = LiveActionContext(
+                defaults=DefaultsCtx(),
+                tags=self._ctx.tags,
+                env={},  # TODO: Is it correct?
+                inputs=await setup_inputs_ctx(ctx, job.args, action_ast.inputs),
+            )
+            job = action_ast.job
+        assert isinstance(job, ast.Job)
+
+        tags = (await self.get_meta(job_id)).tags
+        if job.tags is not None:
+            tags |= {await v.eval(ctx) for v in job.tags}
+
+        env = {}
+        if job.env is not None:
+            env.update({k: await v.eval(ctx) for k, v in job.env.items()})
+
+        title = await job.title.eval(ctx)
+        if title is None:
+            title = f"{self._ctx.flow.flow_id}.{job_id}"
+
+        workdir = (await job.workdir.eval(ctx)) or ctx.defaults.workdir
+
+        volumes = []
+        if job.volumes is not None:
+            for v in job.volumes:
+                val = await v.eval(ctx)
+                if val:
+                    volumes.append(val)
+
+        life_span = (await job.life_span.eval(ctx)) or ctx.defaults.life_span
+
+        preset = (await job.preset.eval(ctx)) or ctx.defaults.preset
+        port_forward = []
+        if job.port_forward is not None:
+            port_forward = [await val.eval(ctx) for val in job.port_forward]
+
+        return Job(
+            id=job_id,
+            detach=bool(await job.detach.eval(ctx)),
+            browse=bool(await job.browse.eval(ctx)),
+            title=title,
+            name=await job.name.eval(ctx),
+            image=await job.image.eval(ctx),
+            preset=preset,
+            entrypoint=await job.entrypoint.eval(ctx),
+            cmd=await job.cmd.eval(ctx),
+            workdir=workdir,
+            volumes=volumes,
+            life_span=life_span,
+            http_port=await job.http_port.eval(ctx),
+            http_auth=await job.http_auth.eval(ctx),
+            port_forward=port_forward,
+            multi=await self.is_multi(job_id),
+            env=env,
+            tags=tags,
+        )
+
+    @classmethod
+    async def create(cls, config_loader: ConfigLoader) -> "RunningLiveFlow":
+        ast_flow = await config_loader.fetch_flow("live")
+
+        assert isinstance(ast_flow, ast.LiveFlow)
+
+        flow_ctx = await setup_flow_ctx(EMPTY_ROOT, ast_flow, "live", config_loader)
+
+        step_1_ctx = LiveContextStep1(flow=flow_ctx)
+
+        defaults, env, tags = await setup_defaults_env_tags_ctx(
+            step_1_ctx, ast_flow.defaults
+        )
+
+        step_2_ctx = LiveContextStep2(
+            flow=step_1_ctx.flow,
+            defaults=defaults,
+            env=env,
+            tags=tags,
+        )
+
+        # volumes / images needs a context with defaults only for self initialization
+        volumes_ctx = await setup_volumes_ctx(step_2_ctx, ast_flow.volumes)
```

Comment:
Do we need a separate step for volumes/images calculation?
Is `env` or `tags` context available in `volume` spec experessions?

Reply:
We need separate steps of defaults and volumes/images because in batch mode there will be intermediate step of for `args`, but I probably got your question wrong.
The `env` and `tags` contexts are available in `volume` spec expressions - it is the same like current contexts work.

Reply:
It is the implementation detail I think.
How `env`/`tags` can be used by `volume` / `image` calculation? And why?

Reply:
I can imagine something like this:
```
env:
  DATA_DIR: /project/data

volumes:
  data:
    remote: storage:$[[ flow.project_id ]]/data
    mount: ${{ env.DATA_DIR }}
    local: data
```
But it isn't very useful in general. I have no strong opinion here. What is you suggestion?

Reply:
I think `tags` are almost useless in expressions in general, because it is python set and there is no way to access it member in expression.

Reply:
I guess `mount: ${{ args.data_dir }}` can solve the same thing. 
Better start with a stricter system and relax it on demand